### PR TITLE
Add support for non restored state on job submission

### PR DESCRIPTION
**Pull request by YuvalItzchakov**
This PR adds support for providing the `--allowNonRestoredState` flag to a submit job request. If, for some reason, that state isn't persistent between versions of a job, the deployment will fail entirely.

The addition here adds a `allowNonRestoredState` boolean flag to the `crd.yaml`.

### CODE REVIEW

File: pkg/controller/flink/flink.go [hunk 1]
```diff
@@ -245,10 +245,11 @@ func (f *Controller) StartFlinkJob(ctx context.Context, application *v1alpha1.Fl
...
 		client.SubmitJobRequest{
-			Parallelism:   parallelism,
-			SavepointPath: application.Spec.SavepointInfo.SavepointLocation,
-			EntryClass:    entryClass,
-			ProgramArgs:   programArgs,
+			Parallelism:           parallelism,
+			SavepointPath:         application.Spec.SavepointInfo.SavepointLocation,
+			EntryClass:            entryClass,
+			ProgramArgs:           programArgs,
+			AllowNonRestoredState: application.Spec.AllowNonRestoredState,
```
Comment (anandswaminathan) [inline pkg/controller/flink/flink.go]: Add to test that this field is populated.
https://github.com/lyft/flinkk8soperator/blob/master/pkg/controller/flink/flink_test.go#L432

**Comment by anandswaminathan** [comment]
Also you need to run `make generate`

**Comment by YuvalItzchakov** [comment]
Appended a check to the existing test, created a new test to make sure `appendNonRestoredState` is propagated properly when set and ran `make generate`

**Comment by anandswaminathan** [comment]
Great. Thanks @YuvalItzchakov

**Comment by anandswaminathan** [comment]
@YuvalItzchakov This is not the right way to do it.

You will have to plumb the flag through the state machine. This case will not work when there is a rollback scenario.

Check this: https://github.com/lyft/flinkk8soperator/blob/master/pkg/controller/flinkapplication/flink_state_machine.go#L266

You will need to add the flag to the status as well.

**Comment by YuvalItzchakov** [comment]
Thanks @anandswaminathan, I’ll get to work on the fix.

**Pull request merged**